### PR TITLE
arrays-programs_involving_multidimensional_data-medium-q8

### PR DESCRIPTION
**Pull request by KaiKhademi**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [x]  Are the assertion fail messages good?
    - [x]  Logical (does it make sense?)
    - [x]  Readable (can a student understand what’s wrong?)
    - [x]  Not too feed-y (does it give away the answer?)
    - [x]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [x]  Specific (does the fail message relate to the question?)
- [x]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [x]  EOF newlines
- [x]  Do the comments for Parsons follow the correct convention?
- [x]  Is the question being tested sufficiently?
    - [x]  Are there representative cases to test the program?
    - [x]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [x]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”
---
## Website
- [ ] Is the question text on the Gamification document up to date? 
    > Make sure to include the sample output if there is one for the question

### CODE REVIEW

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]
See discord messages

File: src/arrays/programs_involving_multidimensional_data/medium/q8/IdenticalGen.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package arrays.programs_involving_multidimensional_data.medium.q8;
+
+/*
+Write a Java program that creates a random integer in the range of 1 to 1000 (inclusive of both values), and print it.
+Then, the program will also create a 10 * 10 2D array. Next, use a method called squareRoot to
+initialize all the elements of the given array with the square root of the value and print out the elements in the
+2D array.
+
```
Comment (Hedgemon4) [inline src/arrays/programs_involving_multidimensional_data/medium/q8/IdenticalGen.java:8]: Add a sample output

File: src/arrays/programs_involving_multidimensional_data/medium/q8/IdenticalGen.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+Distractors:
+    out[i] = Math.sqrt( size );
+    out[j] = Math.sqrt( size );
+    out[i][j] = Math.round( size );
+    out[i][j] = Math.ceil( size );
+    out[i][j] = Math.floor( size );
+ */
+
+public class IdenticalGen {
+    public static void main(String[] args) {
```
Comment (Hedgemon4) [inline src/arrays/programs_involving_multidimensional_data/medium/q8/IdenticalGen.java:18]: I feel like for questions like this, we should also add some printed strings like "Initialized array: " and then another statement for the modified array.

File: src/arrays/programs_involving_multidimensional_data/medium/q8/IdenticalGen.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package arrays.programs_involving_multidimensional_data.medium.q8;
+
+/*
+Write a Java program that creates a random integer in the range of 1 to 1000 (inclusive of both values), and print it.
+Then, the program will also create a 10 * 10 2D array. Next, use a method called squareRoot to
```
Comment (Hedgemon4) [inline src/arrays/programs_involving_multidimensional_data/medium/q8/IdenticalGen.java:5]: This question mentions a method and there is no method in the solution code.

File: src/arrays/programs_involving_multidimensional_data/medium/q8/MainTest.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    public void runMain() {
+        IdenticalGen.main(new String[0]);
+    }
+
+    @Test
+    public void printsCorrectOutput() {
+        double num = Math.sqrt(Double.parseDouble(getItemByName("num")));
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < n; j++) {
+                CustomAssertions._assertEquals(num, Double.parseDouble(getItemByName(i + " " + j)), "Your program does not print the correct 2D array containing the sqrt of the generated number.");
```
Comment (Hedgemon4) [inline src/arrays/programs_involving_multidimensional_data/medium/q8/MainTest.java:40]: I feel like this error message is almost too specific and could be worded better. Something more like "Your program does not correctly create and print the 2D array based on the generated value." or something similar

File: src/arrays/programs_involving_multidimensional_data/medium/q8/IdenticalGen.java (hunk 3 above)
Comment (Hedgemon4) [reply src/arrays/programs_involving_multidimensional_data/medium/q8/IdenticalGen.java:5]: Can you please add the method and a test for it?

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/arrays/programs_involving_multidimensional_data/medium/q8/MainTest.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                {m, int.class}
+        };
+        MethodTest test = new MethodTest(IdenticalGen.class, "squareRoot", arguments);
+        Object output = test.callMethod();
+        double[][] ans = new double[n][m];
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < m; j++)
+                ans[i][j] = Math.sqrt(num);
+        }
+        CustomAssertions._assertArrayEquals(ans, output, "Your square Root method does not return the correct 2D square root array.");
```
Comment (Hedgemon4) [inline src/arrays/programs_involving_multidimensional_data/medium/q8/MainTest.java:82]:
```suggestion
        CustomAssertions._assertArrayEquals(ans, output, "Your squareRoot method does not return the correct 2D array based on the input parameters.");
```

**Review by Hedgemon4 — APPROVED** [review]
